samcli/lib/telemetry/metrics.py

```diff
@@ -98,6 +99,17 @@ def wrapped(*args, **kwargs):
     return wrapped
 
 
+def sam_boto3_client(*args, **kwargs):
+    user_agent_extra = "aws-sam-{}".format(__version__)
+    if _telemetry_enabled():
+        installation_id = GlobalConfig().installation_id
+        user_agent_extra += installation_id
```

Comment:
should we use another '-' delimiter when you join them? also might be better to just use 
```
'-'.join() instead of +
```